assets/js/blocks/product-on-sale/index.js

```diff
@@ -51,36 +49,21 @@ registerBlockType( 'woocommerce/product-on-sale', {
 		],
 	},
 
+	deprecated: [
+		{
+			attributes: sharedAttributes,
```

Comment:
This isn't the correct attributes object, it needs the `orderby` param added too.